woocommerce/woocommerce-gutenberg-products-block
# Add filters for subtotal string and sale badge

Credit where credit is due: this PR is based on the work done in #3750 by @Aljullu 

The second part of this PR is in the subscriptions repository: 3972-gh-woocommerce/woocommerce-subscriptions

Apart from the changes already added by Albert, this PR will:

- Introduce two new filters:
    - `EXPERIMENTAL_CART_ITEM_SALE_BADGE_FILTER`
    - `EXPERIMENTAL_CART_ITEM_SUBTOTAL_FILTER`
- These will be used in the Cart and Checkout blocks to allow extensions to modify the information displayed in the `ProductPrice` below the item title, and the `SaleBadge`s that appear on the cart lines.
- Update styling for the cart table, specifically it removes the `table-layout: fixed` property and the fixed-width on the total column which will allow the `SaleBadge` component in that column to display without overflowing the container. 


Fixes #3747




### Screenshots
#### Before
![image](https://user-images.githubusercontent.com/5656702/106171210-34d94480-6189-11eb-8eda-3365a9d1da03.png)

#### After
![image](https://user-images.githubusercontent.com/5656702/106171336-52a6a980-6189-11eb-9475-33d70501f1bd.png)



### How to test the changes in this Pull Request:

1. Check out `add/subtotal-price-filter` in subscriptions.
1. Add a non-subscription product to your cart with a sale price.
2. Add a subscription product to your cart with a sale price.
3. Add varying lengths/types of subscription product to your cart, every 2 weeks, every 6 months etc. 
4. Open the cart block and verify the sale badge shows the period for subscriptions products, but not for regular products.
5. Verify the subtotal for subscription products shows under each product name and contains the period.
6. Verify the non-subscription product does not show a period for the subtotal.
7. Check the cart and ensure the billing frequency is present under the product name too.
